### PR TITLE
Add DynamoDbIgnoreNulls to support ignoreNulls in nested beans

### PR DESCRIPTION
**Pull request by zoewangg**
## Description
Add `DynamoDbIgnoreNulls` to support ignoreNulls in nested beans

## Motivation and Context
#2303 

## Testing
Added tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [x] My change requires a change to the Javadoc documentation
- [x] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/mapper/BeanTableSchema.java [hunk 1]
```diff
@@ -203,6 +206,19 @@ private BeanTableSchema(StaticTableSchema<T> staticTableSchema) {
         return builder.build();
     }
 
+    private static AttributeConfiguration resolveAttributeConfiguration(PropertyDescriptor propertyDescriptor) {
```
Comment (cenedhryn) [inline services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/mapper/BeanTableSchema.java:209]: I like this encapsulation of the configuration settings.

**Review by cenedhryn — APPROVED** [review]

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/EnhancedType.java [hunk 2]
```diff
@@ -574,6 +574,10 @@ public boolean equals(Object o) {
             return false;
         }
 
+        if (documentConfiguration != null ? !documentConfiguration.equals(enhancedType.documentConfiguration) :
+            enhancedType.documentConfiguration != null) {
+            return false;
+        }
```
Comment (bmaizels) [inline services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/EnhancedType.java:580]: I assume it's intentional to sneak this bugfix in? Should we mention it in CHANGELOG?

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/mapper/BeanTableSchema.java (hunk 1 above)
Comment (bmaizels) [reply services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/mapper/BeanTableSchema.java:209]: +1

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/mapper/BeanTableSchema.java [hunk 3]
```diff
@@ -265,12 +281,14 @@ private BeanTableSchema(StaticTableSchema<T> staticTableSchema) {
                 return EnhancedType.documentOf(
                     (Class<Object>) clazz,
                     (TableSchema<Object>) ImmutableTableSchema.recursiveCreate(clazz, metaTableSchemaCache),
-                    b -> b.preserveEmptyObject(preserveEmptyObject));
+                    b -> b.preserveEmptyObject(attributeConfiguration.preserveEmptyObject())
+                          .ignoreNulls(attributeConfiguration.ignoreNulls()));
```
Comment (bmaizels) [inline services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/mapper/BeanTableSchema.java:285]: This makes me wonder if we should find a way to just have them share the configuration object. Imagine when we have 20 of these...

**Review by bmaizels — APPROVED** [review]
My comments are all non-binding/blocking.

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/EnhancedType.java (hunk 2 above)
Comment (bmaizels) [reply services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/EnhancedType.java:580]: Actually come to think of it shouldn't we fix hashcode() too? Can we add a regression test for both?

Comment (zoewangg) [reply services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/EnhancedType.java:580]: Good catch! Fixed and added regression tests

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/mapper/BeanTableSchema.java (hunk 3 above)
Comment (zoewangg) [reply services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/mapper/BeanTableSchema.java]: Yeah, created a variable for this

**Pull request merged**